Utilities/StorageFactory/src/RemoteFile.cc

```diff
@@ -52,27 +54,30 @@ int RemoteFile::local(const std::string &tmpdir, std::string &temp) {
   // This is better for grid jobs as the current directory is
   // likely to have more space, and is more optimised for
   // large files, and is cleaned up after the job.
-  if (tmpdir.empty() || tmpdir == ".") {
-    size_t len = pathconf(".", _PC_PATH_MAX);
-    char *buf = (char *)malloc(len);
-    getcwd(buf, len);
+  std::vector<char> temp_chars;
 
-    temp.reserve(len + 32);
-    temp = buf;
-    free(buf);
+  if (tmpdir.empty() || tmpdir == ".") {
+    std::filesystem::path current_path = std::filesystem::current_path();
+    temp_chars.assign(current_path.string().begin(), current_path.string().end());
   } else {
-    temp.reserve(tmpdir.size() + 32);
-    temp = tmpdir;
+    temp_chars.assign(tmpdir.begin(), tmpdir.end());
```

Comment:
Please do `temp_chars.reserve(temdir.size()+30)` before the `assign` so only one allocation has to be done.